src/ui/menu.c

```diff
@@ -267,15 +243,6 @@ static GtkWidget* menu_item_new(MenuItem* menuitem, int workspace_id)
 	{
 		mi = gtk_menu_item_new ();
 	}
-	else if (menuitem->type == MENU_ITEM_IMAGE)
-	{
-		GtkWidget* image = gtk_image_new_from_icon_name(menuitem->stock_id, GTK_ICON_SIZE_MENU);
-
-		mi = gtk_image_menu_item_new();
-
-		gtk_image_menu_item_set_image(GTK_IMAGE_MENU_ITEM(mi), image);
-		gtk_widget_show(image);
-	}
```

Comment:
I have no real opinion here.  It removes a feature, but I don't think it has that much impact. Icons lovers might disagree :)

Reply:
Please do not remove icons from menus. This is one of the special feature of MATE desktop which is missing in gnome and most Mate users want to use until someone like to switch to gtk4.

Reply:
I will not accept removal of icons from menus anywhere in MATE. Re-implementing them locally to work around GTK3 deprecating them (and GTK4 if ever used not supporting them) yes, but removing them no